spanner/changestreams/src/main/java/com/example/spanner/changestreams/ChangeStreamSample.java

```diff
@@ -0,0 +1,255 @@
+    final String databaseId = args[1];
+    final String prefix = args[2];
+    final String tableName = prefix + "_Singers";
+    final String changeStreamName = prefix + "_ChangeStreamSingers";
+    SpannerOptions options = SpannerOptions.newBuilder().build();
+    Spanner spanner = options.getService();
+    DatabaseAdminClient dbAdminClient = spanner.getDatabaseAdminClient();
+    DatabaseClient dbClient = spanner.getDatabaseClient(
+        DatabaseId.of(options.getProjectId(), instanceId, databaseId));
+
+    try {
+      createChangeStream(dbAdminClient, instanceId, databaseId, tableName, changeStreamName);
+      queryChangeStream(dbClient, tableName, changeStreamName);
+      dropChangeStream(dbAdminClient, instanceId, databaseId, tableName, changeStreamName);
+    } catch (Exception e) {
+      System.out.println(e);
+    } finally {
+      spanner.close();
+    }
+  }
+
+  public static void updateDdl(
+      DatabaseAdminClient dbAdminClient, String instanceId, String databaseId,
+      List<String> statements) throws Exception {
+    OperationFuture<Void, UpdateDatabaseDdlMetadata> op = dbAdminClient
+        .updateDatabaseDdl(
+          instanceId,
+          databaseId,
+          statements,
+            null
+        );
+
+    op.get(TIMEOUT_MINUTES, TimeUnit.MINUTES);
+  }
+
+  // [START change_stream_sample_create_change_stream]
+  static void createChangeStream(
+      DatabaseAdminClient dbAdminClient, String instanceId, String databaseId, String tableName,
+      String changeStreamName) throws Exception {
+    System.out.println(
+        String.format("Updating database DDL to create table %s and change stream %s.",
+          tableName, changeStreamName));
+
+    updateDdl(
+        dbAdminClient,
+        instanceId,
+        databaseId,
+        Arrays.asList(
+          String.format("CREATE TABLE %s ("
+            + "  SingerId   INT64 NOT NULL,"
+            + "  FirstName  STRING(1024),"
+            + "  LastName   STRING(1024)"
+            + ") PRIMARY KEY (SingerId)", tableName),
+          String.format("CREATE CHANGE STREAM %s FOR %s", changeStreamName, tableName)
+      ));
+  }
+  // [END change_stream_sample_create_change_stream]
+
+  // [START change_stream_sample_query_change_stream]
+  public static void queryChangeStream(DatabaseClient dbClient, String tableName,
+                                       String changeStreamName) {
+    // Insert test data into the table.
+    System.out.println(
+        String.format("Inserting rows "
+          + "(1, Melissa, Garcia) and "
+          + "(2, Dylan, Shaw) "
+          + "into %s table.", tableName));
+
+    insertRows(dbClient, tableName);
+
+    final Timestamp startTimestamp = Timestamp.now();
+    // end = start + 30 seconds.
+    final Timestamp endTimestamp = Timestamp.ofTimeSecondsAndNanos(
+        startTimestamp.getSeconds() + 30, startTimestamp.getNanos());
+    final String startTimestampStr = startTimestamp.toString();
```

Comment:
nit: We should not convert to string here, but do it as late as possible so that we remain using the `Timestamp` type

Reply:
Good point, done.